src/wallet/wallet.cpp

```diff
@@ -1913,8 +1918,10 @@ std::optional<ZTXOSelector> CWallet::ZTXOSelectorForAddress(
                 // determine a local account.
                 auto accountId = this->GetUnifiedAccountId(ufvkMeta.value().GetUFVKId());
                 if (accountId.has_value()) {
-                    if (allowAddressLinkability) {
-                        pattern = AccountZTXOPattern(accountId.value(), ua.GetKnownReceiverTypes());
+                    if (strategy.AllowLinkingAccountAddresses()) {
```

Comment:
I think that passing in the strategy at this point may result in confusing layers of selection; we can see this in the places where we pass in `FullPrivacy` where we're not actually performing any selection at all. The result of this is that those call sites are hard to reason about, because you have to actually look at the implementation here to figure out what's going on. Now that we have switched the other bool to be `TransparentCoinbasePolicy` I think that we should change this back to a boolean value, or even better a separate 2-valued enum.

Reply:
Another thing to consider here is how the "layers of selection" will work in the context of the ZIP 317 wallet implementation. I *think* that what we'll want to do is use the selector to get the full set of notes and/or UTXOs that it's possible for us to spend, and then sub-select from among that set according to both the transaction strategy and the need to pay fees.

Reply:
See https://github.com/zcash/zcash/pull/6377#discussion_r1128838809